Project.toml

```
@@ -12,5 +12,6 @@ Reexport = "189a3867-3050-52da-a836-e630ba90ab69"
 [compat]
 CBinding = "0.9.2"
 MPI = "0.15, 0.16"
+P4est_jll = "~2.3.1"
```

```suggestion
P4est_jll = "= 2.3.1"
```
Do you really want a tilde specifier? If you want to pick a specific version, you need the [equality specifier](https://pkgdocs.julialang.org/v1/compatibility/#Equality-specifier).

---

I assume that if they follow semver at p4est, all patch releases should still work, but since `2.3.0` was not officially released I cannot vouch for it, thus the tilde.

---

Well, promise is maybe too strong, but I think it's a reasonable assumption